### DISCUSSION TITLE
BC setting with Boundary Value Calculation

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#17373] [Q&A General] [answered]
MOOSE Team,
For my transient (unsteady) model, I would like to use NeumannBC for the BC setting and the value of flux is obtained in the following 5 steps for one time level with temperature and time step size called in (total concentration = 0 for the initial condition). Please help.

(1)	For temperature ≤ 100[K], temp = constant_A * temperature
(2)	For temperature > 100[K], temp = constant_B * temperature
then
(3)	concentration at the current time step = temp * ∆t 
(where ∆t = time step size = current time – previous time)
(4)	total concentration = total concentration + temp * ∆t (concentration at the current time step)
(5)	Flux = constant_C * total concentration

Sincerely,
Thomas

**Comment by aeslaughter** [comment, marked as answer]
If I was doing this, I would compute the "total concentration" in a Material object as a material property. Then create a BC object that accepts a constant input parameter and the material property name.

**Reply by styyokuda** [reply]
Thank you Andrew!
I am still new and learning MOOSE.  Please be patient with me.
Are you suggesting to modify source code "Material.C" in /moose/framework/src/materials to include my 5 steps to calculate "total concentration?"
I was thinking to use "FunctionNeumannBC" with "ParsedFunction" so that I can put my 5 steps in the input file, which is a lot easier for me. Is this possible?
Please let me know.
Sincerely,
Thomas

**Reply by aeslaughter** [reply]
No. You should never need to modify source code of MOOSE. You should be working within your own custom application. I suggest you work through our tutorial: https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial01_app_development/index.html

**Reply by styyokuda** [reply]
Thank you Andrew!
I apologize for my confusing question.
My question is what is the specific Material object you suggest to use (GenericConstantMaterial, GenericFunctionMaterial, or Material.C" in /moose/framework/src/materials).
I do appreciate your patience with me.
Sincerely,
Thomas

**Reply by aeslaughter** [reply]
You would start by creating an object that inherits from `Material`, as done here: https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial01_app_development/step09_mat_props.html#step-9-develop-a-material-object
